src/kernel/src/memory/page.rs

```diff
@@ -12,6 +12,11 @@ impl<'a> VPages<'a> {
         Self { mm, ptr, len }
     }
 
+    pub fn add(&mut self, offset: usize) {
+        self.ptr = unsafe { self.ptr.add(offset) };
+        self.len -= offset;
+    }
+
```

Comment:
This will cause a memory leak because `munmap` will not apply to the front of allocation.

Reply:
...What do I do then? XD

Reply:
You can allocate the stack with the original size (not adding an additional page) and just change the protection on the last page without adjusting the pointer to skip the guard page.

Reply:
...Yeah my brain ain't working here.

Reply:
I tried to see what you were getting at, check last commit XD